src/ripple/consensus/Consensus.h

```diff
@@ -1108,6 +1135,113 @@ Consensus<Adaptor>::phaseOpen()
     }
 }
 
+template <class Adaptor>
+bool
+Consensus<Adaptor>::shouldPause() const
+{
+    auto const& parms = adaptor_.parms();
+    std::uint32_t const desynced (previousLedger_.seq() -
+        std::min(adaptor_.getValidLedgerIndex(), previousLedger_.seq()));
+    auto quorumKeys = adaptor_.getQuorumKeys();
+    auto const& quorum = quorumKeys.first;
```

Comment:
Be aware of cases that `quorum` goes to maxint in some cases to protect the validator from forking.

Reply:
ledgerMAX_CONSENSUS is set to 10s. The maximum time we'll pause.

Reply:
I still think its worth checking that case explicitly to avoid doing math that doesn't act as you would expect. It seems like something that could bite you when trying to analyze why a node acted in a certain way.

Reply:
I think desired behavior when quorum is maxint and all the other criteria are met, is for a pause, just like as if the quorum was normal but with all of the other criteria having been met. The debug log output displays all of the variables going into the pause/no-pause decision, including quorum.

Should the desired behavior if quorum is maxint but all of the other criteria have been met (including presence of laggards) to be to not pause?

Reply:
I'm ok with pausing when the quorum is maxint.

My concern is the robustness of this code over the long term. Suppose you or someone else in the future decides to do some additional tweak that depends on the value of quorum, but which doesn't decay gracefully when quorum is maxint. I'd rather future-proof against that by explicitly handling the special case now.

The underlying problem is that we are using maxint for a quorum at all, but changing that seems even more out of scope.

Consider this a suggestion for improvement and not a hard requirement.